Bump envoy_reader to 0.8.6, fix missing dependency

jesserizzo opened the pull request:
## Description:
Bump envoy_reader to 0.8.6, which fixes missing requests_async dependency

**Related issue (if applicable):** fixes #25331

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [ ] There is no commented out code in this PR.
  - [ ] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [ ] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

cgtobi committed:
Add Netatmo climate battery level (#25143)

* Add battery level sensor

* Only update battery level if lower or nonexistent

cgtobi committed:
Fix Netatmo climate battery level (#25165)

* Interpolate battery level

* Sort list

balloob committed:
Restore sensiobo turn on/off methods (#25321)

fredrike committed:
Restore Daikin A/C on/off services (#25332)

jesserizzo committed:
Bump envoy_reader to 0.8.6, fix missing dependency

jesserizzo committed:
Bump envoy_reader to 0.8.6, fix missing dependency

MartinHjelmare commented:
requests-async, which is a dependency of envoy_reader, requires Python 3.6. We still need to maintain Python 3.5.3+ compatibility for this release cycle, and our CI isn't update yet.

dmulcahey committed:
add available to device info (#25349)

dmulcahey committed:
add available to device info (#25349)

scop committed:
Fix util.ruamel_yaml type errors (#25338)

scop committed:
Mypy config improvements (#25340)

* Specify python version

So that it type checks against the lowest common denominator version,
not the runtime one.

* Disallow incomplete definitions

shbatm committed:
Update Google Maps Location Tracker to use locationsharinglib==4.0.2 (#25316)

* Update google_maps: Bump locationsharinglib to 4.0.2

* Remove unused import.

* Corrections based on review.

farmio committed:
temporary patch to fix KNX climate devices (#25356)

This is a temporary patch for knx climate devices. It should be reverted when #24738 is merged to release. 
It should fix https://github.com/home-assistant/home-assistant/issues/25247 for 0.96

cgtobi committed:
Fix preset service call (#25358)

cgtobi committed:
Fix preset service call (#25358)

schmic committed:
Fix for hvac_modes list being null (#25347)

* Fix for empty hvac_modes list

* Empty list instead of default value for hvac_modes

schmic committed:
Fix for hvac_modes list being null (#25347)

* Fix for empty hvac_modes list

* Empty list instead of default value for hvac_modes

balloob committed:
Updated frontend to 20190720.0

balloob committed:
Updated frontend to 20190720.0

bachya committed:
Automatically expand WWLLN window to 1 hour (if necessary) (#25357)

* Expand default window for WWLLN

* Fleshed out conditions

* Fixed tests

* Removed unused import

* Linting

balloob committed:
Updated frontend to 20190721.0

balloob committed:
Updated frontend to 20190721.0

dmulcahey committed:
change and condition to or condition (#25374)

dmulcahey committed:
change and condition to or condition (#25374)

scop committed:
Type check all helpers (#25373)

* Type check all helpers, add inline exclusions for work in progress

* Remove unused Script._template_cache

* Add some missing type hints

* Remove unneeded type: ignore

* Type hint fixes

* Mypy assistance tweaks

* Don't look for None in deprecated config "at most once" check

* Avoid None name slugify attempt when generating entity id

* Avoid None state store attempt on entity remove

dmulcahey committed:
fix remove and re-add scenario (#25370)

dmulcahey committed:
fix remove and re-add scenario (#25370)

OttoWinter committed:
Fix ESPHome climate migration (#25366)

OttoWinter committed:
Fix ESPHome climate migration (#25366)

dmulcahey committed:
bump quirks version (#25362)

dmulcahey committed:
bump quirks version (#25362)

zxdavb committed:
Improve geniushub logging and bump client (#25359)

* add debug logging

* bump geniushub client library

* delint

* bump again

* bump again, again

zxdavb committed:
Improve geniushub logging and bump client (#25359)

* add debug logging

* bump geniushub client library

* delint

* bump again

* bump again, again

Commit:
Update zwave climate mappings (#25327)

hvac_action should be idle when thermostat is in Pending Heat or Pending Cool.

Commit:
Update zwave climate mappings (#25327)

hvac_action should be idle when thermostat is in Pending Heat or Pending Cool.

scop committed:
Optional and Union simplifications (#25365)

balloob committed:
Introduce PRESET_NONE for climate (#25360)

* Introduce PRESET_NONE for climate

* Require preset mode to be a string

* Lint

* Fix tests

balloob committed:
Introduce PRESET_NONE for climate (#25360)

* Introduce PRESET_NONE for climate

* Require preset mode to be a string

* Lint

* Fix tests

balloob committed:
Bumped version to 0.96.3

balloob committed:
Merge pull request #25376 from home-assistant/rc

0.96.3

balloob committed:
Updated frontend to 20190721.1

balloob committed:
Updated frontend to 20190721.1

postlund committed:
Add add_torrent service to Transmission (#25144)

* Add add_torrent service to Transmission

* Fix services.yaml format

* Verify that torrent is whitelisted

* Add logging if adding failed

* Change warn to warning

dwradcliffe committed:
Add support for contact binary sensors in homekit_controller (#25355)

zxdavb committed:
Tweak evohome migration (#25281)

* Initial commit

add hvac_action to zones, remove target_temp from controller

fix incorrect hvac_action

de-lint

Initial commit

de-lint & minor refactor

tweak docstrings, and type hints

tweak docstrings

* refactor setpoints property

* tweak docstring

* tweak docstring

* avoid a unnecessary I/O

* avoid unnecessary I/O

* refactor schedule/setpoints

* remove type hint

* remove type hint 2

* tweak code

* delint type hints

* fix regression

zxdavb committed:
Tweak evohome migration (#25281)

* Initial commit

add hvac_action to zones, remove target_temp from controller

fix incorrect hvac_action

de-lint

Initial commit

de-lint & minor refactor

tweak docstrings, and type hints

tweak docstrings

* refactor setpoints property

* tweak docstring

* tweak docstring

* avoid a unnecessary I/O

* avoid unnecessary I/O

* refactor schedule/setpoints

* remove type hint

* remove type hint 2

* tweak code

* delint type hints

* fix regression

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Jc2k committed:
Only poll HomeKit connection once for all entities on a single bridge/pairing (#25249)

* Stub for polling from a central location

* Allow connection to know the entity objects attached to it

* Move polling logic to connection

* Don't poll if no characteristics selected

* Loosen coupling between entity and HKDevice

* Disable track_time_interval when removing entry

* Revert self.entities changes

* Use @callback for async_state_changed

* Split out unload and remove and add a test

* Test that entity is gone and fix docstring

amelchio committed:
Update pysonos to 0.0.22 (#25399)

amelchio committed:
Update pysonos to 0.0.22 (#25399)

zxdavb committed:
[climate] Correct evohome hvac_action (#25407)

* inital commit

* take account of rounding up of curr_temp

zxdavb committed:
[climate] Correct evohome hvac_action (#25407)

* inital commit

* take account of rounding up of curr_temp

bachya committed:
Add area support to fan service schemas (#25409)

bachya committed:
Add area support to cover service schemas (#25408)

* Add area support to cover service schemas

* Linting

bachya committed:
Add area support to automation service schemas (#25403)

* Add area support to automation service schemas

* Fixed import

* Mispelling

bachya committed:
Add area support to counter service schemas (#25401)

* Add area support to counter service schemas

* Updates

bachya committed:
Add area support to alarm_control_panel service schemas (#25402)

* Add area support to alarm_control_panel service schemas

* Corrected import

zxdavb committed:
[climate] Bugfix/Tweak honeywell migration (#25369)

* refactor supported_features

add dr_data

delint

simplify code

refactor target_temps, and delinting

delint

fix typo

add min/max temps

delint

refactor target temps

refactor target temps 2

correct typo

tweak

fix potential bug

fix potential bug 2

bugfix entity_id

fix typo

* remove explicit entity_id

* refactor hvac_action

* refactor hvac_action

* bugfix - HVAC_MODE_HEAT_COOL incorrectly added to hvac_modes

zxdavb committed:
[climate] Bugfix/Tweak honeywell migration (#25369)

* refactor supported_features

add dr_data

delint

simplify code

refactor target_temps, and delinting

delint

fix typo

add min/max temps

delint

refactor target temps

refactor target temps 2

correct typo

tweak

fix potential bug

fix potential bug 2

bugfix entity_id

fix typo

* remove explicit entity_id

* refactor hvac_action

* refactor hvac_action

* bugfix - HVAC_MODE_HEAT_COOL incorrectly added to hvac_modes

cgtobi committed:
Clean up Netatmo sensor code (#25390)

* Clean up code

* Add parameter

* Make it a list

* Move loop and add debug message

* Further clean up

* Move manual config

cgtobi committed:
Bump pyatmo version to 2.1.2 (#25296)

cgtobi committed:
Bump pyatmo version to 2.1.2 (#25296)

bachya committed:
Add area support to script service schemas (#25439)

bachya committed:
Add area support to scene service schemas (#25438)

bachya committed:
Add area support to input select service schemas (#25432)

bachya committed:
Add area support to input number service schemas (#25431)

bachya committed:
Add area support to input datetime service schemas (#25430)

* Add area support to input datetime service schemas

* Fixed tests

bachya committed:
Add area support to Wink service schemas (#25445)

bachya committed:
Add area support to input boolean service schemas (#25429)

bachya committed:
Add area support to image processing service schemas (#25428)

bachya committed:
Add area support to input text service schemas (#25434)

bachya committed:
Add area support to climate service schemas (#25441)

* Add area support to climate service schemas

* Incorrect Voluptuous usage

* Fix typo

JXGA committed:
Upgrade HPILO requirement to v4.3 (#25444)

* Update Python-HPILO to 4.3

Update of Python-HPILO requirement to 4.3 to resolve outstanding SSL connections for older HP servers (ILO 3)

* Update requirements_all.txt

Update HPILO to 4.3

andre-richter committed:
Increase vallox robustness on startup (#25382)

* Vallox: Increase robustness on startup

Experiments showed that timing of websocket requests to the Vallox firmware is
critical when fetching new metrics. Tests on different Raspberry Pis and x86
machines showed that those machines with little processing power tend to fail
the timing requirments during the busy startup phase of Home Assistant,
resulting in the Vallox integration failing to set itself up.

This patch catches Websocket's InvalidMessage, which is a symptom of failing the
timing requirements. Experiments again showed that on the Raspberry's, this
exception is catched once at startup, but the integration is running fine
afterwards.

* Update __init__.py

* Bump to new 2.1.0 version of api.

* Bump to api 2.2.0

webdjoe committed:
Add Support for VeSync Devices - Outlets and Switches (#24953)

* Change dependency to pyvesync-v2 for vesync switch

* async vesync component

* FInish data_entry_flow

* Update config flow

* strings.json

* Minor fix

* Syntax fix

* Minor Fixs

* UI Fix

* Minor Correct

* Debug lines

* fix device dictionaries

* Light switch fix

* Cleanup

* pylint fixes

* Hassfest and setup scripts

* Flake8 fixes

* Add vesync light platform

* Fix typo

* Update Devices Service

* Fix update devices service

* Add initial test

* Add Config Flow Tests

* Remove Extra Platforms

* Fix requirements

* Update pypi package

* Add login to config_flow

Avoid setting up component if login credentials are invalid

* Fix variable import

* Update config_flow.py

* Update config_flow.py

* Put VS object into hass.data instead of config entry

* Update __init__.py

* Handle Login Error

* Fix invalid login error

* Fix typo

* Remove line

* PEP fixes

* Fix change requests

* Fix typo

* Update __init__.py

* Update switch.py

* Flake8 fix

* Update test requirements

* Fix permission

* Address change requests

* Address change requests

* Fix device discovery indent, add MockConfigEntry

* Fix vesynclightswitch classs

* Remove active time attribute

* Remove time_zone, grammar check

bachya committed:
Add area support to utility meter service schemas (#25442)

* Add area support to utility meter service schemas

* Reverted mistaken deletion

pattyland committed:
Add Elgato Avea integration (#24281)

* Adds Elgato Avea integration

* Revert "Adds Elgato Avea integration"

This reverts commit 8607a685eb47e99ea0e8a69fbdb84e511aaf1be0.

* Adds Elgato Avea integration

* Removed debug

* Improved readability

Co-Authored-By: Otto Winter <otto@otto-winter.com>

* Adds Elgato Avea integration

* Fixes for flake8

* More Fixes for flake8

* Hopefully last fixes for flake8

* Unnecessary rounding and typo removed

* Duplicate calls removed

* raise PlatformNotReady if communication with bulb failes

* Fixes: flake8, missing import of exception and better handling of ble problems

* Update requirements_all.txt

Add requirements_all.txt file

* Revert "Update requirements_all.txt"

This reverts commit 2856025ed3d88bb081e3a8d0fb065f1e7ba1ef45.

* Update requirements_all.txt

* conform with snake_case naming style

https://circleci.com/gh/home-assistant/home-assistant/31823

* Fixed variable rename

* Unnecessary calculation removed

Co-Authored-By: Otto Winter <otto@otto-winter.com>

* Better Exception Handling

* Changed position of import, renamed add_entities to add_devices, remove unnecessary comment

* Unnecessary comments removed.

bachya committed:
Add area support to lock service schemas (#25435)

* Add area support to lock service schemas

* extend

* Fixed tests

* Fixed tests

bachya committed:
Add area support to remote service schemas (#25437)

* Add area support to remote service schemas

* Base

ooii committed:
Suez water (#23844)

* Add suez water sensor

* flake8 test

* pylint test

* edition to fix flake8 and pylint issues

* edition to be okay with the musts

* Added a blank line to __init.py__ for flake8

* added blank line for flake8

* changer scan interval from 10 to 720 minutes

* use of pysuez

* bug fix and isort

* use of pysuez

* fixed flake8 and pylint errors

* update requirements_all.txt

* added a method to test login/password befire adding device

* flake8 edition

* update requirements_all.txt

* add of .coveragerc file with untested files

* update of .coveragerc

* Update homeassistant/components/suez_water/__init__.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Update homeassistant/components/suez_water/sensor.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Update homeassistant/components/suez_water/sensor.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Update homeassistant/components/suez_water/sensor.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Update homeassistant/components/suez_water/sensor.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Update homeassistant/components/suez_water/sensor.py

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* bug fix in check credentials

* flake8 and pylint fixes

* fix codeowner

* update requirements_all.txt

* Sorted suez_water line

* edition to answer comments from @MartinHjelmare on #23844

* Attribute keys formatting to lowercase snakecase, name and icon constants returned directly, and remove of  attribute. Update of .

* pylint edition

* correction wrong keys in client attributes

* remove of unnedeed return and move add_entities

kimfrellsen committed:
Add new device tracker supporting Fortinet FortiGate (#23078)

* Create device_tracker.py

initial version

* Update device_tracker.py

set verify SSL to false as default. Normally users do not have a verified certificate at home

* Update device_tracker.py

pep8 compliant

* Update device_tracker.py

upgraded fortiosapi requirements

* Create __init__.py

* tox compliant

* Update device_tracker.py

* Create manifest.json

* Update .coveragerc

added fortios

* Update device_tracker.py

circle ci, blank line required

* Update manifest.json

removed code owners

* Update manifest.json

removed dependencies

* Update manifest.json

removed codeowners

* Update requirements_all.txt

added fortios

* Update requirements_all.txt

* Update device_tracker.py

pylint corrections

* Update device_tracker.py

pylint exceptions

* Update device_tracker.py

disable pylint broad exceptions

* Update device_tracker.py

pylint

* Update device_tracker.py

removed pointless string statements

* Update device_tracker.py

removed blank line

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update device_tracker.py

* Update manifest.json

added codeowners

* Update CODEOWNERS

added kimfrellsen as codeowner

* fortiosapi 0.10.8

Updated to use latest version of fortiosapi 0.10.8

* Update requirements_all.txt

updated fortiosapi to 0.10.8

* Update device_tracker.py

fixed some requests.

* Update device_tracker.py

better exception handling.

* Update device_tracker.py

exception handling

* Update CODEOWNERS

* Update device_tracker.py

corrected exception handling

* Update device_tracker.py

exception handling.

* Update device_tracker.py

lint corrections

* Update device_tracker.py

removed broad exception.

* Update device_tracker.py

fix lint errors

* Update device_tracker.py

minor changes, mostly cosmetic

fredrike committed:
Update Daikin preset modes (#25395)

* fix preset documentation

* Use pydaikin set holiday method

fredrike committed:
Update Daikin preset modes (#25395)

* fix preset documentation

* Use pydaikin set holiday method

balloob committed:
Bumped version to 0.96.4

Adminiuga committed:
Bump up ZHA dependencies. (#25450)

bachya committed:
Add area support to media player service schemas (#25436)

* Add area support to media player service schemas

* Re-establish MEDIA_PLAYER_SCHEMA

* Comment

* Localize platforms that used MEDIA_PLAYER_SCHEMA

balloob committed:
Merge pull request #25452 from home-assistant/rc

0.96.4

scop committed:
Huawei LTE misc improvements (#25377)

* Rename traffic_statistics to monitoring_traffic_statistics

For better consistency with huawei-lte-api.

* Add default device name for sensors

In case the actual device name cannot be accessed for some reason.

* Support device class in sensor metadata

* Mark known signal strength sensors as such

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

bachya committed:
Add area support to timer service schemas (#25440)

* Add area support to timer service schemas

* Base

* Remove unnecessary TIMER_SERVICE_SCHEMA

bachya committed:
Remove unnecessary REMOTE_SERVICE_SCHEMA (#25453)

bachya committed:
Add area support to vacuum service schemas (#25443)

* Add area support to vacuum service schemas

* Fixed tests

* De-couple platform schemas

fredrike committed:
Daikin simplification and code cleanup (#25416)

* fix preset documentation

* Use pydaikin set holiday method

* update temperature readings, code simplification

* more temperature cleanup

* cleanup HVAC_MODE

* remove get() method and move code to respectivly place

* remove string constant in code

* remove get() method and move code to respectivly place

* isort results

* fixes in state method

zxdavb committed:
Bump geniushub client (#25458)

zxdavb committed:
Bump geniushub client (#25458)

plafue committed:
Allow configuring sources for older Pioneer receivers (#25305)

* Allow configuring sources for older Pioneer receivers

* Replace config.get calls with dict[key] syntax

bachya committed:
Add area support to group service schemas (#25410)

scop committed:
Lint fixes (#25462)

scop committed:
Install requirements_test.txt for flake8 in Azure CI (#25463)

Was missing specified versions, as well as flake8-docstrings altogether.

Santobert committed:
Fix bloomsky unit system (#25460)

* initial commit - fix bloomsky unit system

* Add another warning

* Fix linting error

* Include metric sensor units

* Shorten a too long line

scop committed:
Type check homeassistant.scripts (#25464)

* Run mypy on homeassistant.scripts, disabling bunch of checks for now

* Declare async_initialize in AuthProvider

* Add some type hints

* Remove unreachable code

* Help mypy out

* Script docstring fixes

balloob committed:
Fix Nest turning off eco (#25472)

* Fix Nest turning off eco

* Add hvac action

balloob committed:
Fix Nest turning off eco (#25472)

* Fix Nest turning off eco

* Add hvac action

scop committed:
Mypy config cleanups (#25475)

* Move file specific config to inline comments

* Disallow untyped defs by default everywhere

balloob committed:
Allow cors for static files (#25468)

balloob committed:
Allow cors for static files (#25468)

michaeldavie committed:
Add scan interval to config of Environment Canada sensor (#25414)

* Make refresh period configurable

* Remove throttle, set SCAN_INTERVAL

* Move SCAN_INTERVAL to module

Kane610 committed:
UniFi block clients (#25478)

* Allow blocking clients on UniFi networks

Kane610 committed:
Add a unique identifier to deCONZ groups (#25485)

* Add a unique identifier to deconz groups

balloob committed:
Bumped version to 0.96.5

balloob committed:
Merge pull request #25486 from home-assistant/rc

0.96.5

aschamberger committed:
Add ord() to template filters (#25398)

* Add ord() to template filters

* Remove trailing whitespace

* add test

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

squishykid committed:
Update solax to 0.1.2 (#25497)

bollewolle committed:
Add De Lijn (Flemish Public Transport) component (#24265)

* Initial commit of De Lijn (Flemish Public Transport) component

* Code corrections as per HA dev's requests

* changes to variable naming, setting attribution and states, plus some smaller optimizations

* Overlooked some linting issues, these are now fixed

* Updated pydelijn version requirement to 0.5.1 so UTC timestamps can be used instead of relative/local times.
Removed unused definition

* Updated pydelijn version requirement to 0.5.1 in requirements_all.txt

* Update the self._attributes dict directly instead of replacing it
Assign ATTRIBUTION while creating the _attributes dict
Remove the ATTRIBUTION assignment in device_state_attributes as it's updated in the async_update now.

* Linting issue (lenght of 2 lines) solved

* Removed a relative time attribute
Updated a linting issue in the LOGGER (used % instead of the format)

pvizeli committed:
[skip ci] Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

balloob committed:
Fix python 3.5 test

balloob committed:
Fix deprecation warning in test (#25506)

cgarwood committed:
Improve handling of Z-Wave config entry vs yaml config (#25112)

* Improved handling of config entry vs yaml config

* Address review comment

balloob committed:
Fire lovelace updated event when update detected (#25507)

gjbadros committed:
Support multiple Elk instances (#23839)

* Support multiple Elk instances

* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

QUESTION: Should the "devices" section be optional to avoid breaking
old configuration files?  I chose not to do that for simplicity and
because I was following the doorbird code which requires the "devices"
section for all configurations even with only one device.

* Fixed a bunch of hound-raised issues

Fixed issues raised by hound -- there was clearly
a tool I was supposed to run to get those warnings
before submitting the PR.  Sorry!

Updated REQUIREMENTS.

* Fixed whitespace and line-length mistakes

Also fixed unused prefix local variable lint warning.

* Fixed missing blank line

* Fixed more lint warnings.

Not sure if I missed these on the first pass or if the linter stopped
after a certain number of warnings or something else.

Switched logging to use %d and %s instead of string concatenation (per
lint request and because I imagine it migth be better performing
in some (oldish, I presume) implementations of python.

* Fixed typo in last commit.

* Eliminate devices subsection in config schema

This eliminates the breaking change for configurations wanting a
singleton elk m1 instance (the majority of users, no doubt).  I did
not do it like this before because I was following the lead of the
doorbird component which introduced a devices: section when moving
to support multiple doorbells.  But Rohan Kapoor kindly pointed me
at the zoneminder component which sets the other (IMO) preferable
precedent. Will update the docs change shortly.

* Call async_add_entities once for all the elk controllers.

Just move async_add_entities() outside of the loops across the elk m1
controllers, so it's called once for each platform.

* Call async_add_entities only once per platform.

Move it to after the loop, so it's called only once
per platform even when there are multiple elk m1 controllers.

* Various improvements to be more idiomatic python + bug fixes

Thanks to Martin Hjelmare for the careful review and suggestions.
(All mistaken improvements and new bugs are my own.)

* Removed semicolon that lint caught.

* Idiomatic python improvements

Use dict.values() (instead of making it easier to add local looping variable
on the keys by using _, bar = ...items())

Use [] when the key is known to exist.

* Support multiple Elk instances

* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

QUESTION: Should the "devices" section be optional to avoid breaking
old configuration files?  I chose not to do that for simplicity and
because I was following the doorbird code which requires the "devices"
section for all configurations even with only one device.

* Fixed a bunch of hound-raised issues

Fixed issues raised by hound -- there was clearly
a tool I was supposed to run to get those warnings
before submitting the PR.  Sorry!

Updated REQUIREMENTS.

* Fixed whitespace and line-length mistakes

Also fixed unused prefix local variable lint warning.

* Fixed missing blank line

* Fixed more lint warnings.

Not sure if I missed these on the first pass or if the linter stopped
after a certain number of warnings or something else.

Switched logging to use %d and %s instead of string concatenation (per
lint request and because I imagine it migth be better performing
in some (oldish, I presume) implementations of python.

* Fixed typo in last commit.

* Eliminate devices subsection in config schema

This eliminates the breaking change for configurations wanting a
singleton elk m1 instance (the majority of users, no doubt).  I did
not do it like this before because I was following the lead of the
doorbird component which introduced a devices: section when moving
to support multiple doorbells.  But Rohan Kapoor kindly pointed me
at the zoneminder component which sets the other (IMO) preferable
precedent. Will update the docs change shortly.

* Call async_add_entities once for all the elk controllers.

Just move async_add_entities() outside of the loops across the elk m1
controllers, so it's called once for each platform.

* Call async_add_entities only once per platform.

Move it to after the loop, so it's called only once
per platform even when there are multiple elk m1 controllers.

* Various improvements to be more idiomatic python + bug fixes

Thanks to Martin Hjelmare for the careful review and suggestions.
(All mistaken improvements and new bugs are my own.)

* Removed semicolon that lint caught.

* Idiomatic python improvements

Use dict.values() (instead of making it easier to add local looping variable
on the keys by using _, bar = ...items())

Use [] when the key is known to exist.

* Use dict[key] instead of .get (incl. fixing typo). Use .values() instead of .items() when ignoring keys.

* Gotta use devices.get(prefix) since we use no prefix for the singleton elk instance

* fix requirement to use newer elkm1 that supports my changes for multiple elk devices

* Removed spurious + between a string broken between two lines for formatting; was failing a lint check about logging needing to use %s

* Remove REQUIREMENTS and DEPENDENCIES since those are now taken care of by the manifest.json file.

* Add configuration check that the prefixes are all unique

* Use new dependency 'getmac' to get mac address of Elk M1 controllers and use that for uniqueid if possible, else use None.  Also removed some procedural checking of unique prefix since that's now handled at schema check time.

* Whitespace changes to make style checker happy and code more consistent

* Removed unused variable, added blank line

* Make getmac a requirement not dependency

I should've RTFM.

* ws only change; I really need to get Emacs to understand these style guidelines

* Ran script/gen_requirements_all.py; script/setup needed to be run so that was failing.

* More style check fixes and one bug fix.

* Incomplete set of changes from last push

* More conform-to-hass-style changes: use caps to start log message (and do not use function name even for debug message. And do not use string concatenation; prefer new-style .format.

* Style fixes.

* Switch back to using the prefix config field for setting the unique_id since the mac address approach has numerous shortcomings including: 1) new dependency; 2) lack of reliability; 3) doesn't work for serial connections; 4) breaks when a layer 4+ networking entity intermediates the elk m1 connection.

* Reran to update (removing getmac dependency)

* Skipped trailing ','; keep forgetting which languages are forgiving about this practical nicety of allowing trailing commas without changing the semantics.

* Validate uniqueness on lowercase versions of the prefix since we're gonna use .lower() on creating the entity id that has to be unique; do the _has_all_unique_prefixes check last so we get errors from the device schema before complaining about the uniqueness problem, if any

* Use vol.Lower to convert to lowercase instead of the map.  Also fixed a pair of bugs for the alarm control panel display message service -- since data templates always generate strings, the values subject to range/set restrictions need to be coerced to their proper type before the check

* Fix some flake8 warnings.

* Fixed typo; it's Coerce not coerce.

* Use elkm1m_ string to start unique_id when and only when there is a non-empty prefix given; this enables backward compatibility to avoid a breaking change by letting the elkm1_ start to unique_id keep working exactly as it used to.

* minor comment tweak to force automation tests to run again since they failed for unrelated reasons last time

* There's actually been a 0.7.15 release which was meta-information and tidying only so we might as well depend on it

* Forgot to update this with gen_requirements_all.py

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

oncleben31 committed:
Add container settings for YAML extension to avoid Hass specific custom tags errors (#25504)

* Add settings for YAML extension to avoid !secret tag errors

* add other HA custom tags.

manonstreet committed:
Add last_run_success boolean attribute to Switchbot for error trapping (#25474)

* Add last_run_error boolean attribute to Switchbot entity to allow for trapping of errors

* Add last_run_success boolean attribute to Switchbot for error trapping

* Add last_run_success boolean attribute to Switchbot for error trapping

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

Kane610 committed:
deCONZ - Add power attribute for consumption sensors (#25512)

* Add power attribute for consumption sensors

* Bump dependency to v62

thomasgermain committed:
Improve seventeentrack (#25454)

* Code improvement + tests

* review

* review + moving to pytest test function

* move test to async

* remove code comment

Anglac committed:
Roombalocate (#25508)

* Add Roomba Locate

* Update vacuum.py

Kane610 committed:
deCONZ - cleanup sensor attributes (#25540)

* Improve attributes

mjj4791 committed:
Update to buienradar json api; and additional monitored_conditions  (#24463)

* Update to json api; and additional sensors

* remove unneeded coordinate rounding

* minor optimizations

* Update CODEOWNERS

* Update sensor.py

unit conversion for:
- windgust (ms to km/h)
- windspeed (m/s to km/h)
- windspeed_1d (ms/km/h)
- visibility (m to km)

* Update weather.py

unit conversion for windspeed (m/s to km/h)

* Update sensor.py

* Update weather.py

* Update weather.py

* Update CODEOWNERS

* Update manifest.json

* Update CODEOWNERS

* Update CODEOWNERS

yet another try to get CODEOWNERS accepted...

* update codeowners

* update codeowners

* update

* updated codeowners to match manifest

bachya committed:
Fix bug with WWLLN update interval (#25498)

* Fix bug with WWLLN update interval

* Match the docs

cameronrmorris committed:
Fix WinkAC mode API calls to correct methods (#25545)

Commit:
Make cool_on and heat_on method calls. They aren't properties in python-wink (#25549)

michaeldavie committed:
Bump env_canada to 0.0.19 (#25548)

* Bump env_canada to 0.0.19

* Update requirements_all.txt

Danielhiversen committed:
Tibber, off peak values (#25320)

* Tibber, off peak values

* style

* style

* refactor

* style

cereal2nd committed:
Enable velbus config entries (#25308)

* Initial work on config_flow

* Finish config flow

* Pylint checks, make sure the import only happens once

* Added support for unloading, small fixes

* Check in the hassfest output files

* Flake8 fixes

* pylint mistake after flake8 fixes

* Work on comments

* Abort the import if it is already imported

* More comments resolved

* Added testcases for velbus config flow

* Fix pylint and flake8

* Added connection test to the config flow

* More sugestions

* renamed the abort reason

* excluded all but the config_flow.py from the velbus component in coveragerc

* Rewrote testcases with a patched version of _test_connection

* Docstyle fixes

* Updated the velbus testcases

* just yield

* flake8 fixes

scop committed:
huawei_lte: try unsupported data retrievals only once (#25524)

* huawei_lte: try unsupported data retrievals only once

Refs https://github.com/home-assistant/home-assistant/pull/23809

* Move huawei_lte_api imports to top level

scop committed:
Ignore .dmypy.json (#25528)

amelchio committed:
Update eternalegypt to 0.0.8 (#25551)

fbradyirl committed:
Update Cisco Mobility Express module version (#25422)

* Update requirements_all.txt

* Update manifest.json

* Bump to remove f-strings

* Bump to remove f-strings

Commit:
Expose last_video_id as property for Ring camera (#25553)

andersonshatch committed:
Quiet noisy tado query logging (#25529)

* Quiet noisy tado query logging

* Lint fix

ktnrg45 committed:
Add PS4 tests for media player (#25415)

* Add tests for media_player

* remove ps4/media_player.py

* Add unsubscribe method to unload

* Add_to_hass instead of add_to_manager

* Use hass.states for states

* Fix assertions

* fix tests

* Add schedule update

* Remove entity assertions

Kane610 committed:
UniFi POE control restore clients (#25558)

* Restore POE controls on restart

alengwenus committed:
LCN cover control via output ports (#25511)

* LCN motor control via oputput ports

* Remove default value from cover validator

tofuSCHNITZEL committed:
updated telegram trusted IPs (#25564)

updated telegram trusted IPs (Source: https://core.telegram.org/bots/webhooks)

Kane610 committed:
UniFi device tracker restore clients (#25532)

bachya committed:
Transition SimpliSafe data retrieval to its own object (#25546)

* Transition SimpliSafe data retrieval to its own object

* Don't overwrite a variable

* Member comments

* Member comments

Kane610 committed:
UniFi - Track devices (#25570)

JonGilmore committed:
pylutron PyPI update (#25557)

* pylutron PyPI update

We've been working with the original maintainer of pylutron, and they've published an update to PyPI to support a couple different things: homeowner keypads, main repeater keypads

* added requirements

cereal2nd committed:
Make the velbus component more robust in handling errors (#25567)

* Add some try excepts for velbus as the python-velbus lib is not good in handling these

* Only catch velbusExceptions

* only wrap the lines that can cause the exception

* Fix indentation mixup

orson1282 committed:
Updated Workday Binary Sensor to use Holidays 0.9.11 and added support for Aruba Holidays. (#25568)

* Updated Workday Binary Sensor to use Holidays 0.9.11 and added support for Aruba Holidays.

* updated requirements_all.txts

benleb committed:
add cleaning state code for roborock s6 (#25500)

Adminiuga committed:
ZHA log helper (#25543)

* Logging helper.
* Use log helper for ZHA entities.
* Use log helper for ZHA core device.
* Log helper for ZHA core channels.
* Lint
* ZHA fixture fix.

Quentame committed:
Fix : Velbus translation error (#25575)

balloob committed:
Fix Ecobee HVAC action + available presets (#25488)

* Fix Ecobee HVAC action + available presets

* Update python-ecobee-api to 0.0.21

* Include proper operation list.

* Allows pass on preset to set_climate_hold

* Remove aux heat as a preset

* Fix test

bachya committed:
Add last event data (including "changed_by") to SimpliSafe (#25569)

* Add "last event" sensor for SimpliSafe

* Functionality round 1

* Cleanup

* Whitespace

* Whitespace

* Updated requirements

* Removed unused constants

* Member comments

balloob committed:
Bump Python support to min Python 3.6.0 (#25582)

* Bump Python support to min Python 3.6.0

* Fix type

balloob committed:
Add Black

teharris1 committed:
Bump insteonplm to 0.16.5 (#25580)

JonGilmore committed:
Fix status of lutron switches/lights after HA reboot (#25592)

manutenfruits committed:
Add support for Roku TVs to be powered on or off (#25590)

michaeldavie committed:
Bump env_canada to 0.0.20 (#25594)

iamtpage committed:
Bump venstarcolortouch to 0.9 (#25585)

* Update manifest.json

* Update requirements_all.txt

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

elupus committed:
Log platform import errors and correct reqs for config check (#25425)

* Log failures to load plaforms

* Drop unused exception variable

* Also load pip requirements with check config

* Fix lint

* More lint

* Drop invalid parameters for error call

DarkFox committed:
Improve and align Rejseplanen with other transport components (Breaking) (#25375)

* Improve and align Rejseplanen with other transport components (Breaking)

* Fix empty list check

* Remove pointless list cast

* Clean up redundant definition of transport types

* Add docstring

* Simplify _times check

bachya committed:
Add migration notification for Ambient PWS (#25561)

* Add migration notification for Ambient PWS

* Remove unnecessary kwarg

Co-Authored-By: Andrew Sayre <6730289+andrewsayre@users.noreply.github.com>

* Automatically recreate config entry

* Delete entities and devices

* Move imports

croghostrider committed:
Fix wrong exposed light for emulated hue  (#25581)

* Add auto detect if brightness is supported

* Fix

* Fix tests

* Cleanup

zxdavb committed:
Fix bug and bump geniushub client (#25599)

Fix bug, delint and bump client

tofuSCHNITZEL committed:
implemented timout setting for telnet switch (#25602)

ChristianKuehnel committed:
remove myself from CODEOWNDERS (#25593)

* remove myself from CODEOWNDERS

Sorry, I do not have time right now to work on HomeAssistant, so I'm removing myself from the CODEOWNERS.

* Update manifest.json

rossdargan committed:
Change how ring polls for changes to allow more platforms to be added (#25534)

* Add in a switch to control lights and sirens

* Improve the way sensors are updated

* fixes following flake8

* remove light platform, and fix breaking test.

* Resolve issues with tests

* add tests for the switch platform

* fix up flake8 errors

* fix the long strings

* fix naming on private method.

* updates following p/r

* further fixes following pr

* removed import

* add additional tests to improve code coverage

* forgot to check this in

yeralin committed:
Introduce support for color temperature (#25503)

* Introduce support for color temperature

* Fix linter errors

* Remove extra whitespace for pylint

cgtobi committed:
Netatmo climate refactor (#25457)

* Refactor climate component to use home id rather than name

* Bump pyatmo version

* Add new exception

* Update pyatmo version

balloob committed:
Bump frontend to 20190731.0

thomasloven committed:
Return history for entities in the order they were requested (#25560)

* Return history for entities in the order they were requested

* Fix problems. Add tests

* Update __init__.py

scop committed:
Python < 3.6 remainder cleanups (#25607)

scop committed:
Azure flake8 dep, docstring fixes (#25605)

* Fix Azure CI flake8 deps

* Docstyle fixes

balloob committed:
Black

balloob committed:
Lint

balloob committed:
Lint

balloob committed:
Type

balloob committed:
Make sure typing matches stdlib

balloob committed:
Uninstall typing

balloob committed:
Confirm deletion

balloob committed:
Add constraints to key files

balloob committed:
Merge pull request #25584 from home-assistant/black

Black

balloob committed:
Add translations

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Version bump to 0.98.0.dev0

dmulcahey committed:
bump quirks (#25618)

scop committed:
script/test: fix tox env (#25620)

scop committed:
Move mypy settings to setup.cfg (#25611)

pvizeli committed:
Remove liner for editor. Not need with black

tleegaard committed:
Add support for HomeKit CO2 sensors (#25603)

elupus committed:
Fix test_install_existing_package (#25627)

homeassistant.util.package.install_package is not
a corutine.

Kane610 committed:
Handle disabled devices (#25625)

scop committed:
Blacken top level *.py (#25621)

* Blacken top level *.py

* Tolerate double quotes too in setup.py dependency extraction

Jc2k committed:
Fix polling HomeKit devices with multiple services per accessory (#25629)

scop committed:
Huawei LTE sensor unique id improvements (#25609)

* Convert sensor setup to async

* Improve sensor unique ids

* Save some indent levels, use f-string formatting

* Require getmac in tests

* Fix RouterData init in tests

* Make discovery_info optional in async_setup_platform signature

balloob committed:
Filter out empty results in history API (#25633)

balloob committed:
Expose comfort presets as HA presets (#25491)

* Expose comfort presets as HA presets

* Fix bugs

* Handle unavailable

* log level debug on update

* Lint

scop committed:
Add some debugging to azure mypy job (#25632)

Jc2k committed:
Bump homekit_python to 0.15 (#25631)

oncleben31 committed:
Meteofrance improve log error messages (#25630)

* Improve log error messages

* remove unique_id not ready yet

DarkFox committed:
Fix handling of empty results from Rejseplanen (#25610)

* Improve handling of empty results from Rejseplanen (Fixes #25566)

* Exclude attributes with null value

* Add period back into docstring

* Fix formatting

balloob committed:
Upgrade hass-nabucasa to 0.16 (#25636)

nielstron committed:
Add each fronius sensor as own template (#25608)

* Fix formatting in history test

* Add each sensor as own template

* Make adapters non-entities

* Externalize and organize data fetching, improve system view

* Small fixes

Rename fetching adapters to adapters
throw away non-working system overviews
slightly change naming
remove scan_interval from schema
formatting

* Scan interval is already timedelta and unnecessary return

* Formatting

* Ensure better codestyle by storing cell variables explicitely in different places

CoMPaTech committed:
Add Plugwise component (#25533)

* Added Plugwise component

* pylint fixes

* Additional pylints (local tox not noticing these)

* Changes according to review

* CI flake8 correction

* Applying Black

* Review changes, without exception and schema

* Review changes, exception improvement

* Review changes, exception correctio and schema cleanup

* Further cleaning as per review

* Removed blank line

* Add debugging to API init

* Add debugging to API init without dot

balloob committed:
Updated frontend to 20190801.0

balloob committed:
Add preset to be away and eco (#25643)

Commit:
Add HmIP-SCI to Homematic IP Cloud, Fix HmIP-SWDM (#25639)

* Add HmIP-SCI to Homematic IP Cloud

* Bump upstream dependency

* Fix HmIP-SWDM

cereal2nd committed:
Fix velbus codeowner (#25649)

Santobert committed:
Feature zwave preset modes (#25537)

* Initial commit

* Add some more code

* Local tests passing

* Remove unnecessary line

* Add preset attributes to __init__

* Remove some more debugger lines

* Add some tests

* Fix comparision to None

* Improve test coverage

* Use unknown modes as presets

* Bugfixes and test improvements

* Add tests for unknown preset modes

* linting

* Improve mappings

* Move PRESET_MANUFACTURER_SPECIFIC to zwave

* Replace isinstance with cast

* Add test for hvac_action

* hvac_mode is never None

* Improved mapping of current mode to hvac/preset modes

* Fix bugs where hvac_mode is None

* Add default hvac mode

* Fixed default hvac mode

* Fix linting

* Make flake happy

* Another linting

* Make black happy

* Complete list of default hvac modes

* Add mapping to heat/cool eco

* Fixed another bug where mapping goes wrong

Kane610 committed:
UniFi - allow configuration to not track clients or devices (#25642)

* Allow configuration to not track clients or devices

Adminiuga committed:
ZHA code cleanup. (#25644)

* isort ZHA imports.

* Sort zha channel registry.

* Sort ZHA core registry.

* Sort ZHA core consts.

johnnychicago committed:
Add ELECTRICITY_IMPORTED_TOTAL obis to dsmr (#25655)

Adminiuga committed:
isort ZHA imports. (#25660)

scop committed:
Azure mypy related tweaks (#25663)

* Revert "Add some debugging to azure mypy job (#25632)"

This reverts commit 767b8e9f25affbebb98d65f57c0d78b532a951ba.

No longer needed.

* Install setup.py dependencies for mypy in Azure

For better coverage, and to match what tox does.

amelchio committed:
Revert flux_led to 0.89 (#25653)

* Revert Black

* Revert "Introduce support for color temperature (#25503)"

This reverts commit e1d884a4841f853bb4247a76fca417c0b4f4e7c1.

* Revert "Fix flux_led only-white controllers (#22210)"

This reverts commit 48138189b3c24261fe62a78b6ec854c761d5ce63.

* Revert "Fix MagicHome LEDs with flux_led component (#20733)"

This reverts commit 1444a684e02fab99648f6e5daea9f28b6cf45c10.

* Re-Black

* Use mode detection for scanned bulbs

scop committed:
Spelling fixes (#25666)

Kane610 committed:
Options to not track wired clients (#25669)

jesserizzo commented:
@MartinHjelmare Assuming that lack of Python 3.5 support is the only thing failing this PR, will it be able to be merged in time for the HA 0.97 release?

MartinHjelmare commented:
We have now update our CI with minimum Python version at 3.6.

Please rebase on latest dev branch to let the build pass.

Commit:
Add PRESET_AWAY to HomematicIP Cloud climate (#25641)

* enable climate away_mode and home.refresh

* Add Party eco modes

jesserizzo committed:
Merge branch 'dev' of github.com:jesserizzo/home-assistant into dev

jesserizzo commented:
Sorry! I screwed that up.
